### DISCUSSION TITLE
Passing a value from Dart to Rust `std::sync::mpsc::channel` doesn't do anything

### DISCUSSION THREAD

**Discussion by temeddix** [fzyzcjy/flutter_rust_bridge#1065] [Q&A] [answered]
Passing a value from Dart into `std::sync::mpsc::channel` with FRB generated code doesn't do anything. If I make a loop in a separate thread in Rust and put in a value to `std::sync::mpsc::channel`, it does work. However, when I call Rust function in `api.rs` from Dart and put in a value in `std::sync::mpsc::channel`, it doesn't work at all. Does this has to do something with Dart isolate?

**Comment by temeddix** [comment]
I think this was because of circular dependency

**Comment by fzyzcjy** [comment]
No, I guess it is a bug in your code? Maybe make a minimal reproducible sample

**Comment by temeddix** [comment, marked as answer]
I made a mistake of calling library file(crate) from Dart two times, that caused global variables to be initialized twice in Rust side 😅
